simperby/src/node/history_storage.rs

```diff
@@ -52,7 +52,16 @@ async fn get_json_from_storage<T>(storage: &dyn KVStorage, key: Hash256) -> Resu
 where
     T: serde::de::DeserializeOwned,
 {
-    let value = storage.get(key).await?;
+    let value = match storage.get(key.clone()).await {
+        Ok(x) => x,
+        Err(simperby_kv_storage::Error::NotFound) => {
+            return Err(Error::StorageIntegrityError(format!(
```

Comment:
Is this function always expected to take a key that should be in the storage?

Reply:
@lemonholic Probably not, but is NotFound error insufficient for such cases?

Reply:
clarified in dd9ac7e

Reply:
got it